src/main/java/edu/ucsb/mapache/controllers/SearchHistoryController.java

```diff
@@ -58,25 +58,38 @@ public DecodedJWT getJWT(String authorization) {
     }
 
     @GetMapping("/allusersearches")
-    public ResponseEntity<String> getSearches(@RequestHeader("Authorization") String authorization)
+    public ResponseEntity<String> getalluserSearches(@RequestHeader("Authorization") String authorization)
             throws JsonProcessingException {
         if (!authControllerAdvice.getIsMember(authorization))
             return getUnauthorizedResponse("member");
         
-       if (!authControllerAdvice.getIsAdmin(authorization)) {
+      
+          if (!authControllerAdvice.getIsAdmin(authorization)) {
+             return getUnauthorizedResponse("admin");
+          }
+        
+        
+        Iterable<UserSearch> usersearch = usersearchRepository.findAll();
+        String body = mapper.writeValueAsString(usersearch);
+        return ResponseEntity.ok().body(body);
+    }
+    
+    @GetMapping("/specificuser")
+    public ResponseEntity<String> getspecificSearches(@RequestHeader("Authorization") String authorization)
```

Comment:
I think a better name for this endpoint and method would be `/mySearches` and `getMySearches`, since you can't arbitrarily ask for any user's searches.